dotCMS/src/main/java/com/dotcms/security/apps/AppsAPIImpl.java

```diff
@@ -68,26 +69,69 @@
             .setVisibility(PropertyAccessor.FIELD, Visibility.ANY)
             .findAndRegisterModules();
 
-    private String toJsonString(final AppSecrets object) throws DotDataException {
+    /*
+    char[] toJsonAsChars(final AppSecrets object) throws DotDataException {
         try {
-            return jsonMapper.writeValueAsString(object);
+            final byte [] bytes = jsonMapper.writeValueAsBytes(object);
+            return bytesToCharArrayUTF(bytes);
         } catch (IOException e) {
             throw new DotDataException(e);
         }
     }
 
-    private AppSecrets readJson(final String json) throws DotDataException {
+    AppSecrets readJson2(final char[] chars) throws DotDataException {
         try {
-            return jsonMapper.readValue(json, AppSecrets.class);
+            final byte [] bytes = charsToBytesUTF(chars);
+            return jsonMapper.readValue(bytes, AppSecrets.class);
         } catch (IOException e) {
             throw new DotDataException(e);
         }
-    }
+    }*/
 
     private AppSecrets readJson(final char[] chars) throws DotDataException {
-        return readJson(String.valueOf(chars));
+        try {
+            return jsonMapper.readValue(new String(chars), AppSecrets.class);
+        } catch (IOException e) {
+            throw new DotDataException(e);
+        }
+    }
+
+    private String toJsonAsString(final AppSecrets object) throws DotDataException {
+        try {
+            return jsonMapper.writeValueAsString(object);
+        } catch (IOException e) {
+            throw new DotDataException(e);
+        }
     }
 
+    /**
```

Comment:
Same as above.

Reply:
yeah sorry for leaving it. I'm gonna use it soon.